### PR TITLE
feat: correct address house number domain

### PR DESCRIPTION
**Pull request by jvandaal**

### CODE REVIEW

File: src/AddressRegistry/StreetName/StreetNameAddress.cs [hunk 1]
```diff
@@ -378,6 +379,44 @@ private void CorrectPostalCodeBecauseParentPostalCodeWasChanged(PostalCode posta
...
+            {
+                throw new AddressHasInvalidStatusException();
+            }
+
+            if (BoxNumber is not null)
+            {
+                throw new HouseNumberToCorrectHasBoxNumberException();
+            }
+
+            var formatRegex = new Regex("^[1-9]([0-9]{0,8}([A-H]|[K-N]|[P]|[R-T]|[V-Z]){0,1}|[0-9]{0,9})$", RegexOptions.Compiled);
```
Comment (ArneD) [inline src/AddressRegistry/StreetName/StreetNameAddress.cs:398]: move the regex somewhere central (static)?
Also should we also do this check in propose?

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/AddressRegistry/StreetName/Events/AddressHouseNumberWasCorrectedV2.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            StreetNamePersistentLocalId = streetNamePersistentLocalId;
+            BoxNumberPersistentLocalIds = boxNumberPersistentLocalIds.Select(x => (int)x).ToList();
+            HouseNumber = houseNumber;
+        }
+
+        [JsonConstructor]
+        private AddressHouseNumberWasCorrectedV2(
+            int streetNamePersistentLocalId,
+            int addressPersistentLocalId,
+            string houseNumber,
```
[the remark this answers is not in the document]
Comment (jvandaal) [reply src/AddressRegistry/StreetName/Events/AddressHouseNumberWasCorrectedV2.cs]: Done

File: src/AddressRegistry/StreetName/StreetName.cs [hunk 3]
```diff
@@ -298,6 +298,18 @@ public void CorrectAddressPostalCode(AddressPersistentLocalId addressPersistentL
                 .CorrectPostalCode(postalCode);
         }
 
+        public void CorrectAddressHouseNumber(AddressPersistentLocalId addressPersistentLocalId, HouseNumber houseNumber)
+        {
+            if (StreetNameAddresses.HasActiveAddressForOtherThan(houseNumber, addressPersistentLocalId))
```
[the remark this answers is not in the document]
Comment (jvandaal) [reply src/AddressRegistry/StreetName/StreetName.cs:311]: Done

File: src/AddressRegistry/StreetName/StreetNameAddress.cs (hunk 1 above)
Comment (jvandaal) [reply src/AddressRegistry/StreetName/StreetNameAddress.cs]: Done

**Pull request merged**